### PR TITLE
Agent Auto Auth Self Healing for Templates

### PR DESCRIPTION
**Pull request by divyaac**
This PR adds the ability for Agent to re-trigger auto auth, when consul-template recieves an "invalid token" error while rendering a template.

Link to ENT PR: https://github.com/hashicorp/vault-enterprise/pull/5627

### CODE REVIEW

File: changelog/26172.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:feature
+agent: Agent will re-trigger auto auth if token used for rendering templates has been revoked, has exceeded the number of uses, or is a bogus value.
```
Comment (VioletHynes) [inline changelog/26172.txt:2]: We should change this from `feature` to `improvement` or use the feature name syntax, e.g.
`**Agent Auto-Auth Self-Healing**: ...`
To me I think this makes more sense as an improvement (customer requires no changes, it affects old features), but it's up to you!

**Review by VioletHynes — COMMENTED** [review]
Apologies, I thought the previous PR I was reviewing was a CE PR. I noticed none of the files were enterprise only, and these affect the same 21 files--what's the difference in the diff? This one has -41 lines and the other had -44.

**Comment by divyaac** [comment]
> Apologies, I thought the previous PR I was reviewing was a CE PR. I noticed none of the files were enterprise only, and these affect the same 21 files--what's the difference in the diff? This one has -41 lines and the other had -44.

I believe the diff is in the go.mod and go.sum files. I cleaned it up here to make all the builds pass properly (versus in the enterprise)

**Review by VioletHynes — APPROVED** [review]

**Review by abohossny — APPROVED** [review]
[]()

**Pull request merged**